### PR TITLE
Mongodbfix

### PR DESCRIPTION
**Pull request by Sukhpreet92**

### CODE REVIEW

File: components/Card.js [hunk 1]
```diff
@@ -1,44 +1,49 @@
...
   padding-right: 1em;
-
   &:hover {
     filter: invert(1);
   }
 `;
 
+const CheckBoxIcon = styled.button`
+  border: 0;
+  margin: 0;
```
Comment (PhilippWeiss85) [inline components/Card.js]: if you want to save one line of code you could also write: margin: 0 0.5em 0 0; and kick the margin-right out

File: pages/api/habits/[id].js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    method,
+  } = req;
+
+  await dbConnect();
+
+  switch (method) {
+    case "GET" /* Get a model by its ID */:
+      try {
+        const habit = await Habit.findById(id);
+        if (!habit) {
```
Comment (PhilippWeiss85) [inline pages/api/habits/[id].js:16]: i think this case is already handled by line 17

Comment (PhilippWeiss85) [inline pages/api/habits/[id].js:16]: i think this if statement is already handled by line 21

File: pages/api/habits/[id].js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      }
+      break;
+
+    case "PUT" /* Edit a model by its ID */:
+      try {
+        const habit = await habit.findByIdAndUpdate(id, req.body, {
+          new: true,
+          runValidators: true,
+        });
+        if (!habit) {
```
Comment (PhilippWeiss85) [inline pages/api/habits/[id].js:28]: same as above with line 36

File: pages/api/habits/[id].js (hunk 2 above)
Comment (PhilippWeiss85) [inline pages/api/habits/[id].js]: i think this if statement is already handled by line 21. You should check if this is necessary (i am not sure)

File: pages/api/habits/[id].js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      } catch (error) {
+        res.status(400).json({ success: false });
+      }
+      break;
+
+    case "DELETE" /* Delete a model by its ID */:
+      try {
+        const deletedHabit = await Habit.deleteOne({ _id: id });
+        if (!deletedHabit) {
+          return res.status(400).json({ success: false });
```
Comment (PhilippWeiss85) [inline pages/api/habits/[id].js]: same here

**Review by PhilippWeiss85 — CHANGES_REQUESTED** [review]
Please repair your failed Vercel deployment

**Comment by Sukhpreet92** [comment]
> Please repair your failed Vercel deployment

Hey Thank you for review i have made the corrections accordingly

**Review by PhilippWeiss85 — APPROVED** [review]
lgtm now:)

**Pull request merged**